kernel/idle.c

```diff
@@ -164,6 +164,20 @@ void idle(void *unused1, void *unused2, void *unused3)
 	__idle_time_stamp = (u64_t)k_cycle_get_32();
 #endif
 
+#ifdef CONFIG_SMP
+	/* Simplified idle for non-default SMP CPUs pending driver
+	 * support.  The busy waiting is needed to prevent lock
+	 * contention.  Long term we need to wake up idle CPUs with an
+	 * IPI.
```

Comment:
suggest filing a user story about this and other future improvements, possibly for others to work on as the team has bandwidth.